test/base/FreeSpecDiscipline.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package base
+
+import org.scalactic.Prettifier
+import org.scalactic.source.Position
+import org.scalatest.freespec.AnyFreeSpecLike
+import org.scalatest.prop.Configuration
+import org.scalatestplus.scalacheck.Checkers
+import org.typelevel.discipline.Laws
+import org.typelevel.discipline.scalatest.Discipline
+
+trait FreeSpecDiscipline extends Discipline {
```

Comment:
How does this work?

Reply:
The Discipline trait has a `checkAll` method that you implement for your testing style and it runs the `Laws#RuleSet` that you pass in. There are a few implementations for other testing styles but there wasn't any FreeSpec. The RuleSet hooks in with ScalaCheck to run the tests and uses ScalaCheck's `forAll.